### PR TITLE
kvdo: 8.2.0.2 -> 8.2.1.6; unbreak

### PR DESCRIPTION
**Pull request by erdnaxe**
###### Description of changes

ZHF: #230712

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by erdnaxe** [comment]
Result of `nixpkgs-review pr 231813` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>19 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.kvdo</li>
    <li>linuxKernel.packages.linux_4_14_hardened.kvdo</li>
    <li>linuxKernel.packages.linux_4_19.kvdo</li>
    <li>linuxKernel.packages.linux_4_19_hardened.kvdo</li>
    <li>linuxKernel.packages.linux_5_10.kvdo</li>
    <li>linuxKernel.packages.linux_5_10_hardened.kvdo</li>
    <li>linuxKernel.packages.linux_5_4.kvdo</li>
    <li>linuxKernel.packages.linux_5_4_hardened.kvdo</li>
    <li>linuxKernel.packages.linux_6_1.kvdo</li>
    <li>linuxKernel.packages.linux_hardened.kvdo (linuxKernel.packages.linux_6_1_hardened.kvdo)</li>
    <li>linuxKernel.packages.linux_6_2.kvdo</li>
    <li>linuxKernel.packages.linux_6_3.kvdo</li>
    <li>linuxKernel.packages.linux_latest_libre.kvdo</li>
    <li>linuxKernel.packages.linux_libre.kvdo</li>
    <li>linuxKernel.packages.linux_lqx.kvdo</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.kvdo</li>
    <li>linuxKernel.packages.linux_xanmod.kvdo</li>
    <li>linuxKernel.packages.linux_xanmod_latest.kvdo (linuxKernel.packages.linux_xanmod_stable.kvdo)</li>
    <li>linuxKernel.packages.linux_zen.kvdo</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_15.kvdo</li>
    <li>linuxKernel.packages.linux_5_15_hardened.kvdo</li>
  </ul>
</details>

--- 

This mostly only fixes linux_5_15 module.

**Comment by erdnaxe** [comment]
Result of `nixpkgs-review pr 231813` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.kvdo</li>
    <li>linuxKernel.packages.linux_4_14_hardened.kvdo</li>
    <li>linuxKernel.packages.linux_4_19.kvdo</li>
    <li>linuxKernel.packages.linux_4_19_hardened.kvdo</li>
    <li>linuxKernel.packages.linux_5_10.kvdo</li>
    <li>linuxKernel.packages.linux_5_10_hardened.kvdo</li>
    <li>linuxKernel.packages.linux_5_4.kvdo</li>
    <li>linuxKernel.packages.linux_5_4_hardened.kvdo</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_15.kvdo</li>
    <li>linuxKernel.packages.linux_5_15_hardened.kvdo</li>
    <li>linuxKernel.packages.linux_6_1.kvdo</li>
    <li>linuxKernel.packages.linux_hardened.kvdo (linuxKernel.packages.linux_6_1_hardened.kvdo)</li>
    <li>linuxKernel.packages.linux_6_2.kvdo</li>
    <li>linuxKernel.packages.linux_6_3.kvdo</li>
    <li>linuxKernel.packages.linux_latest_libre.kvdo</li>
    <li>linuxKernel.packages.linux_libre.kvdo</li>
    <li>linuxKernel.packages.linux_lqx.kvdo</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.kvdo</li>
    <li>linuxKernel.packages.linux_xanmod.kvdo</li>
    <li>linuxKernel.packages.linux_xanmod_latest.kvdo (linuxKernel.packages.linux_xanmod_stable.kvdo)</li>
    <li>linuxKernel.packages.linux_zen.kvdo</li>
  </ul>
</details>

**Review by RaitoBezarius — APPROVED** [review]
LGTM; testing it

File: pkgs/os-specific/linux/kvdo/default.nix [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 
   src = fetchFromGitHub {
     owner = "dm-vdo";
     repo = "kvdo";
     rev = version;
-    hash = "sha256-4FYTFUIvGjea3bh2GbQYG7hSswVDdNS3S+jWQ9+inpg=";
+    hash = "sha256-S5r2Rgx5pWk4IsdIwmfZkuGL/oEQ3prquyVqxjR3cO0=";
   };
 
+  nativeBuildInputs = [ pahole ];
```
Comment (RaitoBezarius) [inline pkgs/os-specific/linux/kvdo/default.nix:15]:
```suggestion
  nativeBuildInputs = kernel.moduleBuildDependencies;
```

File: pkgs/os-specific/linux/kvdo/default.nix [hunk 2]
```diff
@@ -1,16 +1,19 @@
-{ stdenv, lib, fetchFromGitHub, vdo, kernel }:
+{ stdenv, lib, fetchFromGitHub, vdo, kernel, pahole }:
```
Comment (RaitoBezarius) [inline pkgs/os-specific/linux/kvdo/default.nix:1]:
```suggestion
{ stdenv, lib, fetchFromGitHub, vdo, kernel }:
```

File: pkgs/os-specific/linux/kvdo/default.nix [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
 
 stdenv.mkDerivation rec {
-  inherit (vdo) version;
+  inherit (vdo);
   pname = "kvdo";
+  version = "8.2.1.6"; # bump this version with vdo
```
Comment (RaitoBezarius) [inline pkgs/os-specific/linux/kvdo/default.nix:6]: Why the version are decorrelated now?

**Review by RaitoBezarius — CHANGES_REQUESTED** [review]
Please use `kernel.moduleBuildDependencies`

**Pull request merged**